### ISSUE TITLE
Rule S1172: False Positive with array of value tuple

### ISSUE THREAD

**Issue by JohnLBevan** [SonarSource/sonar-dotnet#3811]
### Description

I'm seeing the same issue as reported here: https://github.com/SonarSource/sonar-dotnet/issues/1049 when using `dotnet-sonarscanner`.  

### Repro steps

With code such as below:

```csharp

void Main()
{
	Console.WriteLine(PopulateHappyBirthdayTemplate("Alice", "fiftieth"));
}
string PopulateHappyBirthdayTemplate(string name, string age)
{
	return SubstituteTemplatePlaceholders(
		"Hello {name}.\nHappy {age} birthday."
		, ("{name}", name)
		, ("{age}", age)
	);
}
string SubstituteTemplatePlaceholders(string template, params ValueTuple<string, string>[] substitutions)
{
    foreach (var substitution in substitutions)
    {
        template = template.Replace(substitution.Item1, substitution.Item2);
    }
    return template;
}
```

Run Sonar Scanner

```
dotnet-sonarscanner begin \
    /key:"${SonarProjectKey}" \
    /v:"$ProjectVersion" \
    /d:sonar.login="${SonarQubeToken}" \
    /d:sonar.host.url="${SonarHostUrl}" \
    /d:sonar.scm.provider="git" \
    /d:sonar.cs.opencover.reportPaths="${outputDir}/**/coverage.opencover.xml" \
    /d:sonar.cs.vstest.reportsPaths="${outputDir}/**.trx" \
    /d:sonar.qualitygate.wait="${QualityGate}"

dotnet restore "${DotNetSolutionDirectory}" && restore_status=$? || restore_status=$? # continue on error, so the results are pushed to sonarqube

dotnet build "${DotNetSolutionDirectory}" --no-restore && build_status=$? || build_status=$? # continue on error, so the results are pushed to sonarqube

dotnet test "${DotNetSolutionDirectory}" \
    --no-restore \
    --no-build \
    --logger trx \
    --results-directory "${outputDir}" \
    --collect:"XPlat Code Coverage" \
    /p:CollectCoverage=true \
    /p:CoverletOutputFormat=\"opencover,lcov\" \
    /p:CoverletOutput="${outputDir}" \
    && test_status=$? || test_status=$? # continue on error, so the results are pushed to sonarqube

dotnet-sonarscanner end /d:sonar.login="${SonarQubeToken}"
```

### Expected behavior

No issues should be detected.

### Actual behavior

We see warnings for the variables:

> warning S1172: Remove this parameter 'name', whose value is ignored in the method. 
> warning S1172: Remove this parameter 'age', whose value is ignored in the method. 

### Related information

* SonarScanner for MSBuild version: 5.0.2
* Languages in your solution: C#

**Comment by pavel-mikula-sonarsource** [comment]
Hi @JohnLBevan ,

Thank you for reporting this case. I can confirm it as a False Positive. It's not related to arrays, but tuple creations as reported in #3255. We fixed that issue this morning and it will be shipped with next release.
